src/pages/api/opt-out-feedback.ts

```diff
@@ -0,0 +1,118 @@
+import { NextApiRequest, NextApiResponse } from 'next'
+import {
+  GoogleSpreadsheet,
+  GoogleSpreadsheetRow,
+  GoogleSpreadsheetWorksheet,
+} from 'google-spreadsheet'
+
+const FEEDBACK_SHEET_ID = process.env.FEEDBACK_SHEET_ID
+const FEEDBACK_SERVICE_EMAIL = process.env.FEEDBACK_SERVICE_EMAIL
+const FEEDBACK_PRIVATE_KEY = process.env.FEEDBACK_PRIVATE_KEY
+
+export default async function handler(
+  request: NextApiRequest,
+  response: NextApiResponse
+) {
+  // Only allow POST for this endpoint
+  if (request.method !== 'POST') {
+    response.status(404).json({
+      meta: {
+        status_code: 404,
+        status_text: 'Not Found',
+      },
+    })
+    return
+  }
+
+  // Validate Request Headers
+  if (request.headers['content-type'] !== 'application/json') {
+    response.status(400).json({
+      meta: {
+        status_code: 400,
+        status_text: 'Bad Request',
+      },
+      errors: [
+        {
+          msg: 'expects header Content-Type: application/json',
+        },
+      ],
+    })
+    return
+  }
+
+  // Ensure `primary_opt_out_reason` is specified
+  if (!request.body.primary_opt_out_reason) {
+    response.status(400).json({
+      meta: {
+        status_code: 400,
+        status_text: 'Bad Request',
+      },
+      errors: [
+        {
+          msg: 'Required body key `primary_opt_out_reason` is not provided',
```

Comment:
On the form, we have the opt out reason as optional, do we want to make this required?

Reply:
Oh!  Okay yeah so *both* are optional?   We definitely want to treat it as such, then.